src/app/ReadHandler.cpp

```diff
@@ -407,6 +407,26 @@ void ReadHandler::OnResponseTimeout(Messaging::ExchangeContext * apExchangeConte
 {
     ChipLogError(DataManagement, "Time out! failed to receive status response from Exchange: " ChipLogFormatExchange,
                  ChipLogValueExchange(apExchangeContext));
+#if CHIP_CONFIG_ENABLE_ICD_SERVER && CHIP_CONFIG_ENABLE_ICD_CIP
+    switch (mState)
+    {
+    case HandlerState::AwaitingReportResponse:
+        if (IsType(InteractionType::Subscribe) && !IsPriming())
+        {
+            // Delay triggering check-in messages until the ReadHandler is closed, since it remains in the AwaitingReportResponse state,
+            // which blocks check-in messages from being sent.
+            // Additionally, avoid triggering a check-in during subscription setup failures; handling such cases is the client's responsibility.
+            ChipLogError(DataManagement, "Trigger check-in message when non-priming subscription report times out");
+            mManagementCallback.GetInteractionModelEngine()->ScheduledForceTriggerCheckInMessages();
```

Comment:
Depending on the ordering this will force check-in messages to be sent twice if i am not mistaken.
We should only force a check-in for the specific registration and not for all registration.

Also, in the case where the controller is not available, for example during a power outage, the power consumption will be significantly increased due to the back to back operations.

I'm not sure this is an overall improvement on the current implementation...